v2.0.0 source added.

DeividasBakanas opened the pull request.

Commit:
v2.0.0 source added.

GiedriusGrabauskas commented on package.json:
```diff
@@ -0,0 +1,58 @@
+    "bugs": {
+        "url": "https://github.com/SimplrJS/simplr-loaders/issues"
+    },
+    "homepage": "https://github.com/SimplrJS/simplr-loaders#readme",
```
Homepage without #readme

GiedriusGrabauskas commented on example/src/layout/logo.tsx:
```diff
@@ -0,0 +1,25 @@
+import * as React from "react";
+
+export interface Props {
+    imageClass?: string;
```
Rename to className

GiedriusGrabauskas commented on example/src/layout/logo.tsx:
```diff
@@ -0,0 +1,25 @@
+            href="https://simplrjs.com"
+            className="logo-container"
+        >
+            <span className={`logo ${(this.props.imageClass != null ? this.props.imageClass : "")}`}></span>
```
Maybe use classnames dependency?

GiedriusGrabauskas commented on .gitignore:
```diff
@@ -0,0 +1,240 @@
+## Ignore Visual Studio temporary files, build results, and
+## files generated by popular Visual Studio add-ons.
```
Minify gitignore file

GiedriusGrabauskas commented on package.json:
```diff
@@ -0,0 +1,58 @@
+    "devDependencies": {
+        "cleanup-package-json": "^0.2.1",
+        "css-to-ts": "0.1.0-beta.9",
+        "glob-uglifyjs": "0.4.0",
```
Dont uglify ant minify files

GiedriusGrabauskas commented on src/abstractions/loader-base.ts:
```diff
@@ -0,0 +1,124 @@
+    /**
+     * Puts stringified stylesheet of a loader to the `<head>`.
+     * @param styles {string} strigified stylesheet of a loader.
+     */
```
Missing props param in comments

GiedriusGrabauskas replied:
Why dont use this.props?

GiedriusGrabauskas commented on src/bubble-loader/bubble-loader.tsx:
```diff
@@ -0,0 +1,33 @@
+ */
+export class BubbleLoader extends LoaderBase<BaseProps, {}> {
+    constructor(props: BaseProps) {
+        super(props);
```
Missing context

GiedriusGrabauskas commented on src/abstractions/loader-base.ts:
```diff
@@ -0,0 +1,124 @@
+     * Unique identifier of a loader.
+     * Used in `protected AppendStyles` to prevent styles duplications.
+     */
+    abstract LoaderId: string;
```
Wgy public?

GiedriusGrabauskas replied:
Why*

GiedriusGrabauskas commented on src/spinner-loader/spinner-loader.tsx:
```diff
@@ -0,0 +1,33 @@
+ * @extends {LoaderBase<BaseProps, {}>}
+ */
+export class SpinnerLoader extends LoaderBase<BaseProps, {}> {
+    constructor(props: BaseProps) {
```
Missing context

Commit:
`npm run server` -> `npm run example` in example.
PeerDependencies moved to Dependencies package.json.

Commit:
.gitignore minified.
LoaderId changed to protected in loader-base.ts.
Context added to bubble-loader and spinner-loader constructors.
Unnecessary parameter `prop` removed from `AppendStyles` in `loader-base.ts`.
README.md changed to follow changes in source files.
homepage link updated.
`package.json` removed from `files` array.
classNames added to example as a dependency.
imageClass changed to className in logo.tsx of example.

Commit:
2.0.0-beta.1

GiedriusGrabauskas commented on src/abstractions/loader-base.ts:
```diff
@@ -65,14 +65,14 @@ export abstract class LoaderBase<TProps extends BaseProps, TState> extends React
-    protected AppendStyles(styles: string, props: TProps) {
-        if (props.useDefaultStyle === false) {
+    protected AppendStyles(styles: string) {
+        if (this.props.useDefaultStyle === false) {
```
Ammm... `this.props` is not yet set.

Commit:
`AppendStyles` parameters changed in `base-loader`.
README.md updated.

GiedriusGrabauskas commented on .gitignore:
```diff
@@ -0,0 +1,6 @@
+node_modules/
```
Remove `/`

GiedriusGrabauskas commented on CHANGELOG.md:
```diff
@@ -0,0 +1,56 @@
+### [Breaking changes]
+- Directories structure changed. All production files moved to `dist` folder.
+- `blackLoader` prop deprecated in `SpinnerLoader`.
+- `color` prop deprecated in `BubbleLoader`.
```
Update decription to `removed`

GiedriusGrabauskas commented on README.md:
```diff
@@ -1,2 +1,181 @@
+
+export class MyComponent extends React.Component<{}, {}> {
+    render() {
+        return <BubbleLoader/>;
```
Spacing.
```ts
return <BubbleLoader />;
```

GiedriusGrabauskas commented on example/package.json:
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "simplr-loaders-example",
+  "version": "0.1.0",
+  "description": "Example for simplr-loaders.",
```
Add `private: true`

GiedriusGrabauskas commented on package.json:
```diff
@@ -0,0 +1,57 @@
+    "tslint": "tslint --project \"./tsconfig.json\" --c \"./tslint.json\"",
+    "uglifyjs": "glob-uglifyjs",
+    "release": "npm run build && npm run tslint && npm run uglifyjs",
+    "cleanup-package-json": "cpj"
```
Remove cpj

GiedriusGrabauskas commented on package.json:
```diff
@@ -0,0 +1,57 @@
+  },
+  "dependencies": {
+    "react": "^15.3.2",
+    "react-dom": "^15.3.2",
```
react and react-dom move to peerDependencies

Commit:
cleanup-package-json deleted.
CHANGELOG.md updated.
README.md updated.
example changed to be private package.

Commit:
ReactDom updated in example.

DovydasNavickas committed:
Update .gitignore

Commit:
2.0.0-beta.2

Commit:
Merge branch 'feature/v2.0.0' of https://github.com/SimplrJS/simplr-loaders into feature/v2.0.0

DovydasNavickas commented on @types/abstractions/loader-base.d.ts:
```diff
@@ -0,0 +1,88 @@
+     */
+    useDefaultStyle?: boolean;
+    /**
+     * Specifies whether loader should take area of a strict size.
```
Specifies whether the loader should take an area of a strict size.

DovydasNavickas commented on @types/abstractions/loader-base.d.ts:
```diff
@@ -0,0 +1,88 @@
+    /**
+     * Specifies whether loader should take area of a strict size.
+     * By default a loader should expand to take all possible area.
+     * Using this prop you can cancel default behavior.
```
the default behaviour

DovydasNavickas commented on @types/abstractions/loader-base.d.ts:
```diff
@@ -0,0 +1,88 @@
+     * @type {boolean}
+     * @memberOf BaseProps
+     */
+    shouldReduceSize?: boolean;
```
shouldReduceSize -> shouldExpand

DovydasNavickas commented on README.md:
```diff
@@ -1,2 +1,181 @@
-# simplr-loaders
-React css loaders with no prerequisite style loading.
+# simplr-loaders
+React css loaders with no prerequisite style loading.
```
React CSS loaders with no prerequisites for required styles loading.

Commit:
`shouldReduceSize` changed to `shouldExpand`.
CHANGELOG.md and README.md updated.
`description` changed in package.json.

Commit:
2.0.0-beta.3
